test/e2e/auth/audit_dynamic.go

```diff
@@ -0,0 +1,385 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package auth
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+
+	auditregv1alpha1 "k8s.io/api/auditregistration/v1alpha1"
+	apiv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/wait"
+	auditinternal "k8s.io/apiserver/pkg/apis/audit"
+	auditv1 "k8s.io/apiserver/pkg/apis/audit/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/kubernetes/test/utils"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+)
+
+var _ = SIGDescribe("[Feature:DynamicAudit]", func() {
+	f := framework.NewDefaultFramework("audit")
+	BeforeEach(func() {
+		framework.SkipUnlessProviderIs("gce")
```

Comment:
Is DynamicAudit really unsupported with other cloud / cluster providers?

Reply:
According to https://github.com/kubernetes/enhancements/blob/master/keps/sig-auth/0014-dynamic-audit-configuration.md the feature doesn't seem GCE specific. I guess the reason is that this takes care of GCE's configure-helper.sh only at this time.

Reply:
This takes care of GCE's configure-helper.sh only, I guess that is the reason why.

Reply:
So its still in alpha, and from what I could tell the alpha tests only run on GCE. Also, the advanced audit test only runs in GCE, although I'm not sure of the reasoning there.

Reply:
I still don't see that as reason to prevent someone else from trying to set this up or test it against a different cloud provider.  Maybe I want to use cluster-api-provider-aws and enable this feature, this test would prevent me from confirming whether it works.

Please drop this `SkipUnless`